### ISSUE TITLE
[BUG] operations are very slow on mangas with lots of chapters

### ISSUE THREAD

**Issue by Gniiiii** [Paperback-iOS/app#843]
### Preflight Checklist

- [X] I have read and I agree to follow the [support disclaimer](https://github.com/Paperback-iOS/app#support-disclaimer).
- [X] I am sure that my bug has nothing to with third party extensions.
- [X] I have read the [troubleshooting FAQs](https://paperback.moe/help/faq/) and looked trough the \#help-desk channel on the [official Discord](https://discord.com/invite/Paperback), but none solved my issue.
- [X] I have searched the [issue tracker](https://github.com/Paperback-iOS/app/issues) for a bug report that matches the one I want to file, without success.

### What type of Apple device have you encountered the bug with?

iPad

### Operating System Version

iPadOS 17.3

### App Version

0.8.7-r12

### Describe the Bug

I have some mangas with lots of chapters ( a few k).
For these, it takes a long time to display the chapter list, to start reading a chapter or to transition from a chapter to the next.
Additionally, it sometimes happens that chapters are not properly flagged as read and I also suspect that these manga stall the refresh process (but I haven’t how to investigate). 

This is not a network related issue because I have the same symptoms in flight mode with preloaded chapter.
At these times, my iPad grows quite hot, indicating heavy workload.
I tried repairing my database in case but that didn’t change the situation.

I do suspect an indexing strategy with a complexity that does scales well with high number of chapters

### Steps to Reproduce

Find a manga with a lot of chapters (you may need to use a 3rd party source for that…)
Bookmark it
Open the bookmark to display the chapter list
Open a chapter (you can download it first to ensure it’s not affected by network conditions)

### Anything Else?

I noticed an old, closed issue from 2020 that also involve manga with a lot of chapters but lad to a crash during a chapter transition instead. This may have some connection

**Comment by Gniiiii** [comment]
Complementary info: once such big manga is opened, side effects are propagated, sometimes to other manga until the application is forced close and reopened

**Comment by nitsujri** [comment]
Yes this continues to today. Reading Martial Peak is a game of stopping my phone from overheating.

**Comment by celarye** [comment]
As part of a repository-wide cleanup for the 0.9 release, we are resetting our issue tracker to more efficiently track new feedback and feature requests. If this issue is still relevant or critical to your workflow, please feel free to reopen it with updated context. Thank you for your support!
